### DISCUSSION TITLE
Welcome to edit_orders Discussions!

### DISCUSSION THREAD

**Discussion by lat9** [lat9/edit_orders#223] [Announcements]
## 👋 Welcome!
I'm using Discussions as a place to (er) discuss the requirements and wish-lists for the next generation of `Edit Orders`:
  * Many Zen Cart sites don't use `Edit Orders`, why does yours?
  * What features are important to you?
  * How is `Edit Orders` currently used in your site's _workflow_?
 
Note that the next generation of `EO` will support Zen Cart 1.5.8 and later.

**Comment by lat9** [comment]
A couple of things to consider:
1. A product present in the order might no longer be present in the site or its name/model/price might have changed.
   - Need to have a means to identify these discrepancies to the order-editing admin.
   - Give the option (on a product-by-product basis) whether or not to restock the product.
2. Any action taken by EO for the order's actual update will have a confirmation screen so that the admin knows what changes are being committed and has the opportunity to make changes prior to committing those changes.
3. Products will _retain_ their initial `orders_products_id` value on an update; currently products are removed and the order totally rebuilt on an update.
4. The zc158+ Bootstrap/Font Awesome platforms will be used, with (obviously) some screen layout changes.  Bootstrap `modals` will be used to gather form-related inputs and an AJAX handler will handle those form-related updates as well as tracking all changes to the order.
5. There will _most likely_ be downward-incompatible changes.
    - The customer/billing/shipping addresses will be displayed as separate Bootstrap cards instead in a table-based manner.  That initial display will use the basic address-formatting, with a Bootstrap `modal` displayed if an address requires editing.  There are plugins, e.g. [Vat4EU](https://github.com/lat9/vat4eu) that add elements to an address' display.
    - Stock-related handling, e.g. my [Products' Options' Stock Manager](https://vinosdefrutastropicales.com/index.php?main_page=product_info&cPath=2_7&products_id=46), which perform their special stock-related handling based on EO's current adding/removing product observers.  Since products will no longer be removed en mass and re-added, these integrations will require re-validation.

**Comment by highburyeye** [comment]
Yes. Please update this. It is very useful. Some customers want to add products to an order before it is sent, or change parameters for a product. Sometimes its also used for correcting an address typo.
Dosnt seem to work well with Coupons but not a major issue

**Comment by proseLA** [comment]
i have not had time to look at this, and i am not sure if it has been fixed in a recent version of edit orders, so don't shoot me if it has, and if it has not, i am more than willing to look and fix it (when i have time), but in an older version, when i am changing one of the 3 (!!!) addresses, the tab scrolls in the most illogical way.  can we just tab through all of the fields of address 1 first?  and then go to address 2?  etc.  right now it tabs across the 3 names, and then the 3 company names, etc...  i am not sure who works that way, but i do not.  call me radical...

**Reply by lat9** [reply]
OK, "You're radical"!  v4.7.0 updates the order's address-display so that the cursor will traverse the columns top-to-bottom of the left then center then right address' displays.  See "Considerations 5." above.

**Comment by KilgoreLSU** [comment]
Cindy,
If you need some money thrown at this project, please let me know.  
I edit orders all the time.. 90% of my orders are invoiced..  Often my customers order wrong item, or item has been changed, etc.. Lots of reasons to modify.
It would be great if it would interact with Local Sales Tax plugin properly.  I have to zero out everyone tax column/row when I edit an order.  Some of these orders are 20 items+.
Thanks and look forward to this!  I've been rolling over to 1.5.8

**Comment by Matt46750** [comment]
We're on ZC 5.8 and use Edit Orders routinely.  Our site hosts many products which are closely similar, thus many customers end up changing their order (for item, qty or shipping method) after submitting the order.  Being able to edit the order allows the customer to see what actually shipped and provides them an accurate final invoice for the order.   We find it quite helpful.

**Comment by DanTomlinson** [comment]
Why?  Because things change and "Johhny can't type!"  :)
Shipping Address correction would probably be the 3rd-most used - don't really care about the others.
First  sometimes a customer will call to make a change to an order - so sometimes add an item or 2
Second  sometimes we have a stock item missing and are short-shipping them.
Third  an occasional item price change (we'll give a discount on the item)

These can be "big" orders/low amount.   20-50 lines isn't unusual, 
Upgrading to 1.5.8a as a new install from 1.5.5f (which updated from 1.5.0) / bootstrap (modified in progress)

*lat9  Cindy we've talked and you've helped with Beadfx before  :)

**Comment by lat9** [comment]
As I've started into the design phase for EO 5.0.0, there are some questions for which I'd like answers for each site you own or manage that's currently using EO.  EO 5.0.0 is going to be more finicky about whether/not it can edit an order.

If you are uncomfortable providing the answers in this discussion, you can send them to me via my direct email address.  If you don't have that address, you can PM me (lat9) on the Zen Cart forums.

- What version of EO is currently in use?
- What version of Zen Cart?
- Are the site's products all taxed at the same rate?
- Does the site use multiple tax-rates for a given tax-class?
- Does the description of each tax-rate include its tax-percentage, e.g. 'FL Tax 7%'?
- Does the site charge tax on shipping?  If so, does the shipping tax use a different tax-rate than the associated products?
- What order-totals are active on the site?

**Comment by Matt46750** [comment]
See below.  

As I've started into the design phase for EO 5.0.0, there are some questions for which I'd like answers for each site you own or manage that's currently using EO. EO 5.0.0 is going to be more finicky about whether/not it can edit an order.

If you are uncomfortable providing the answers in this discussion, you can send them to me via my direct email address. If you don't have that address, you can PM me (lat9) on the Zen Cart forums.

*	What version of EO is currently in use?[Matt: >] 4.7.1
*	What version of Zen Cart?[Matt: >] 1.5.8a (moving to 2.0 next quarter)
*	Are the site's products all taxed at the same rate?[Matt: >]  No – using two rates: 0% or 7%. 
*	Does the site use multiple tax-rates for a given tax-class?[Matt: >]  No. 
*	Does the description of each tax-rate include its tax-percentage, e.g. 'FL Tax 7%'?[Matt: >]  We use “IN Tax 7%”  and “Sales Tax Exempt Item”.  But could easily add a 0% into that description if helpful. 
*	Does the site charge tax on shipping? If so, does the shipping tax use a different tax-rate than the associated products?[Matt: >]  Affirmative: IN requires the same sales tax on shipping anytime the item is taxable.

**Reply by lat9** [reply]
Yes, adding the '0%" for that 'Sales Tax Exempt Item' will help in the follow-on upgrade.

**Comment by KilgoreLSU** [comment]
Version 4.6 -
Zen 1.5.7c
No items are taxed using Local Sales Tax module due to Louisianas
ridiculous Taxation system.
No.. Taxrate description is not included
No tax on shipping.

I really need it to interact properly with Local Sales Tax module.  If I
edit an order, it does a weird double tax thing I have to zero out and it's
tedious on an order that is 20 to 30 line items.

**Reply by lat9** [reply]
I'll take a look at "Local Sales Tax" offering and see if it's something I can coerce into working correctly with EO.

**Comment by Matt46750** [comment]
You bet.  Changed the description to: “Sales Tax Exempt 0%”

Matt

**Comment by lat9** [comment]
First beta release now available for testing: https://github.com/lat9/edit_orders/releases/tag/v5.0.0-beta1
